lib/restbase.js

```diff
@@ -283,8 +282,18 @@ RESTBase.prototype._request = function(req) {
         }
 
         // Call the handler with P.try to catch synchronous exceptions.
-        var reqHandlerPromise = P.try(handler, [childRESTBase, childReq])
-        .then(function(res) {
+        var reqHandlerPromise;
+        if (handler.validator) {
+            reqHandlerPromise = P.try(function() {
+                return handler.validator.validate(childReq);
+            })
+            .then(function() {
+                return handler(childRESTBase, childReq);
```

Comment:
@d00rman it's already in the promise, so it's equal to the previous functionality. The `P.try` was needed before to 'initialise' a promise chain.